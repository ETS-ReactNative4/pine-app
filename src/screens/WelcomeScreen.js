import React, { Component } from 'react';
import { StyleSheet, Dimensions } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Title from '../components/Title';
import Paragraph from '../components/Paragraph';
import Footer from '../components/Footer';
import Button from '../components/Button';
import Link from '../components/Link';
import BaseScreen from './BaseScreen';

const windowDimensions = Dimensions.get('window');

const styles = StyleSheet.create({
  title: {
    marginBottom: windowDimensions.height < 600 ? 10 : 20
  },
  button: {
    marginBottom: 10
  }
});

@connect()
export default class WelcomeScreen extends Component {
  static navigationOptions = {
    header: null
  }

  _createKey() {

  }

  _recoverKey() {

  }

  render() {
    return (
      <BaseScreen>
        <Title style={styles.title}>
          Welcome to Wallet ID
        </Title>

        <Paragraph>
          Start by creating a new key.
        </Paragraph>

        <Footer>
          <Button
            label='Create a new key'
            loadingLabel='Creating key...'
            onPress={this._createKey.bind(this)}
            style={styles.button}
          />

          <Link onPress={this._recoverKey.bind(this)}>
            Or recover an existing Wallet ID key
          </Link>
        </Footer>
      </BaseScreen>
    );
  }
}

WelcomeScreen.propTypes = {
  dispatch: PropTypes.func,
  navigation: PropTypes.any
};
