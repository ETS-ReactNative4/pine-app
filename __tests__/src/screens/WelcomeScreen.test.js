import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeScreen from '../../../src/screens/WelcomeScreen';

jest.mock('../../../src/containers/BackHeaderContainer', () => 'BackHeaderContainer');

const storeMock = {
  getState: jest.fn(() => ({})),
  dispatch: jest.fn()
};

describe('WelcomeScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <WelcomeScreen store={storeMock} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
