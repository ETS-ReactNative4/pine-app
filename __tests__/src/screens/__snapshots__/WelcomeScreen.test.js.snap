// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flex": 1,
        "justifyContent": "center",
        "padding": 40,
      },
      undefined,
    ]
  }
>
  <HeaderContainer
    backButtonIconStyle={undefined}
    title={undefined}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#26203D",
            "fontSize": 18,
            "fontWeight": "600",
            "letterSpacing": 0.36,
            "marginBottom": 10,
            "textAlign": "center",
          },
          Object {
            "marginBottom": 20,
          },
        ],
      ]
    }
  >
    Welcome to Wallet ID
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#AAAAAA",
            "fontSize": 15,
            "fontWeight": "300",
            "lineHeight": 21,
            "marginBottom": 20,
          },
          Object {
            "textAlign": "center",
          },
        ],
      ]
    }
  >
    One secure key for all your bitcoin wallets and services.
  </Text>
  <View
    pointerEvents={undefined}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "bottom": 30,
          "left": 40,
          "position": "absolute",
          "right": 40,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#322A51",
              "borderRadius": 7,
              "height": 46,
              "justifyContent": "center",
              "shadowRadius": 4,
              "width": 670,
            },
            Object {
              "marginBottom": 10,
            },
          ]
        }
      >
        <ActivityIndicator
          animating={false}
          color="#D2DCFF"
          hidesWhenStopped={true}
          size="small"
          style={
            Object {
              "height": 12,
              "left": 15,
              "position": "absolute",
              "top": 15,
            }
          }
        />
        <RNLinearTextGradient
          accessible={true}
          allowFontScaling={true}
          colors={
            Array [
              4294962687,
              4292663551,
              4292736255,
            ]
          }
          ellipsizeMode="tail"
          gradientEnd={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          gradientStart={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          locations={
            Array [
              0,
              0.5,
              1,
            ]
          }
          style={
            Array [
              Object {
                "color": "gray",
              },
              Object {
                "fontFamily": "System",
                "fontSize": 11,
                "fontWeight": "600",
                "letterSpacing": 0.41,
              },
            ]
          }
        >
          CREATE A NEW KEY
        </RNLinearTextGradient>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Array [
                Object {
                  "color": "#322A51",
                  "fontSize": 12,
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Or import an existing Wallet ID key
        </Text>
      </Text>
    </View>
  </View>
  <ErrorModalContainer />
</View>
`;
